File: ProyectoConRedis/ProbandoRedisConCrud/demo2/src/main/java/com/example/demo/repository/UserRepository.java
package com.example.demo.repository;

import com.example.demo.entity.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Map;

@Repository
public interface UserRepository extends CrudRepository<User, String> {
    User findUsersByNickName(String nickname);

    User findUsersByName(String name);

    User findUsersById(String id);

    User getUserById(String id);

    List<User> findAll();
}

File: kotlin/kotlinSpring/src/main/kotlin/com/joanna/kotlindemo/repository/UserRepository.kt
package com.joanna.kotlindemo.repository

import com.joanna.kotlindemo.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, String> {
    fun findUsersByNickName(nickname:String): User;
    fun findUsersById(id:Long): User;
    fun findUsersByName(name:String): User;
    fun getUserById(id:Long): User;
    override fun findAll(): List<User>;
}